src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import com.google.sps.util.UserHelper;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+
+  // Request: param redirectUrl for after login is finished
+  // Response: if not logged in, redirects to login page -> redirectUrl,
+  //           otherwise redirects to redirectUrl.
+  @Override
+  public void doGet (HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    String redirectUrl = request.getParameter("redirectUrl");
+    if (!userService.isUserLoggedIn()) {
+        String loginPageUrl = userService.createLoginURL(redirectUrl);
+        response.sendRedirect(loginPageUrl);
+        return;
+    }
+
+    response.sendRedirect(redirectUrl);
+    return;
```

Comment:
You can remove this - there's no need to explicitly return at the end of the method